libs/wingsdk/src/target-sim/service.ts

```diff
@@ -0,0 +1,103 @@
+import { join } from "path";
+import { Construct } from "constructs";
+import { Function } from "./function";
+import { ISimulatorResource } from "./resource";
+import { SERVICE_TYPE, ServiceSchema } from "./schema-resources";
+import { simulatorHandleToken } from "./tokens";
+import { bindSimulatorResource, makeSimulatorJsClient } from "./util";
+import * as cloud from "../cloud";
+import * as core from "../core";
+import { IInflightHost, Resource } from "../std";
+import { BaseResourceSchema } from "../testing";
+import { convertBetweenHandlers } from "../utils/convert";
+
+export class Service extends cloud.Service implements ISimulatorResource {
+  private readonly autoStart: boolean;
+  private onStartHandlerToken: string;
+  private onStopHandlerToken?: string;
+
+  constructor(scope: Construct, id: string, props: cloud.ServiceProps) {
+    super(scope, id, props);
+    this.autoStart = props.autoStart ?? true;
+
+    // On Start Handler
+    const onStartHash = props.onStart.node.addr.slice(-8);
+
+    const onStartFunctionHandler = convertBetweenHandlers(
```

Comment:
There's probably some reuse opportunity...

Extract the code from lines 26..48 into a method and reuse in lines 52..75.